src/wallet/rpcwallet.cpp

```diff
@@ -2458,10 +2545,25 @@ UniValue listunspent(const UniValue& params, bool fHelp)
         }
     }
 
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    std::optional<int> asOfHeight;
+    if (params.size() > 3) {
+        asOfHeight = params[3].get_int();
+    }
+
     UniValue results(UniValue::VARR);
     vector<COutput> vecOutputs;
-    LOCK2(cs_main, pwalletMain->cs_wallet);
-    pwalletMain->AvailableCoins(vecOutputs, false, NULL, true);
+    pwalletMain->AvailableCoins(
+            vecOutputs,
+            asOfHeight,
+            false,        // fOnlyConfirmed
+            nullptr,      // coinControl
+            true,         // fIncludeZeroValue
+            true,         // fIncludeCoinBase
+            false,        // fOnlySpendable
+            nMinDepth,
+            destinations.empty() ? nullptr : &destinations);
```

Comment:
Was this incorrect before? If so, document the fix in the release notes.

Reply:
What I meant is that this didn't previously pass `&destinations`.

Reply:
Ok, so I just dug into this a bit. A couple things …
1. I think this change is correct (both `nMinDepth` and `destinations`;
2. I don’t think it needs to be `?:`, empty set & null pointer are handled identically;
3. I think it should be passed by `const &` actually – not sure why this would be a pointer; and
4. I think this obviates lines 2575–2576 (new line numbers) just below this, as `AvailableCoins` has the same code, but for preemptively filtering `vecOutputs`.

I’m going to make these changes.

Reply:
Re: 4., I think you mean [lines 2506-2507 (rhs)](https://github.com/zcash/zcash/pull/6122/files/5ea600a102d6e2e4a062570981d00662ccb41b0e#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dR2506-R2507), but yes I agree.

Reply:
Actually I'm slightly confused by why `CWallet::AvailableCoins` has that parameter since there don't appear to be any uses of it prior to this PR.